src/tools/sample/sample.c

```diff
@@ -994,6 +1003,90 @@ RunClient(
     }
 }
 
+
+//
+// Runs the multi client side of the protocol.
+//
+void
+RunMultiClient(
+    _In_ int argc,
+    _In_reads_(argc) _Null_terminated_ char* argv[]
+    )
+{
+    //
+    // Load the client configuration based on the "unsecure" command line option.
+    //
+    if (!ClientLoadConfiguration(GetFlag(argc, argv, "unsecure"))) {
+        return;
+    }
+    QUIC_STATUS Status;
+    uint32_t NumberOfConnections = 0;
+    HQUIC* Connections = NULL;
+
+    //
+    // Get the target / server name or IP from the command line.
+    //
+    const char* Target;
+    if ((Target = GetValue(argc, argv, "target")) == NULL) {
+        printf("Must specify '-target' argument!\n");
+        Status = QUIC_STATUS_INVALID_PARAMETER;
+        goto Error;
+    }
+
+    const char* NumberOfConnectionsString;
+    if ((NumberOfConnectionsString = GetValue(argc, argv, "count")) == NULL) {
+        printf("Must specify '-count' argument with -multiclient!\n");
+        Status = QUIC_STATUS_INVALID_PARAMETER;
+        goto Error;
+    }
+
+    NumberOfConnections = strtoul(NumberOfConnectionsString, NULL, 10);
+    if (NumberOfConnections > UINT16_MAX) {
+        printf("'-count' parameter %s > 65535!\n", NumberOfConnectionsString);
+        Status = QUIC_STATUS_INVALID_PARAMETER;
+        goto Error;
+    }
+
+    Connections = (HQUIC*)malloc(sizeof(HQUIC) * NumberOfConnections);
+
+    QUIC_CONNECTION_POOL_CONFIG PoolConfig = { 0 };
+    PoolConfig.Registration = Registration;
+    PoolConfig.Configuration = Configuration;
+    PoolConfig.Handler = ClientConnectionCallback;
+    PoolConfig.Family = QUIC_ADDRESS_FAMILY_UNSPEC;
+    PoolConfig.ServerName = Target;
+    PoolConfig.ServerPort = UdpPort;
+    PoolConfig.NumberOfConnections = (uint16_t)NumberOfConnections;
+
+    printf("Connection Pool Connecting...\n");
+
+    //
+    // Start the connections to the server.
+    //
+    if (QUIC_FAILED(Status = MsQuic->ConnectionPoolCreate(&PoolConfig, Connections))) {
```

Comment:
Unless that's something usual in the QUIC codebase, assignment in a if test condition is generally considered error prone.

Reply:
It was used elsewhere in the sample so I followed the convention.